[charts-pro] Release the pro package in alpha

alexfauquette opened the pull request:
## Changelog

- 🎨 Charts get a new component to display color mapping in the legend.
![image](https://github.com/user-attachments/assets/cc8adba5-724c-4954-9527-4de98fdf293a)

- 🚀 The `@mui/x-charts-pro` is released in alpha version 🧪. This new package introduces two main features:
	- The Heatmap component
	- The zoom interaction on the bar, line, and scatter charts

Preview: https://deploy-preview-13859--material-ui-x.netlify.app/x/react-charts/heatmap/

oliviertassinari commented:
Cool to see near its release.

All the things that I can spot on https://deploy-preview-13859--material-ui-x.netlify.app/x/react-charts/heatmap/:

1. `import { UnstableHeatmap } from '@mui/x-charts-pro/Heatmap';` is usually `import { Unstable_Heatmap } from '@mui/x-charts-pro/Heatmap';`
2. `import '@mui/x-charts-pro/typeOverloads';` in the demo seems strange to me. I would expect this to be in the npm package
3. Should https://deploy-preview-13859--material-ui-x.netlify.app/x/react-charts/zoom-and-pan/ also have the "Preveiw" flag? It's released as `7.0.0-alpha.0`
4. https://deploy-preview-13859--material-ui-x.netlify.app/x/introduction/licensing/#pro-plan should also list the charts?
5. The JS file isn't converted

<img width="518" alt="SCR-20240718-bcxn" src="https://github.com/user-attachments/assets/cd57a822-8c0a-4692-8a95-835e42995878">

6. The popover might be able to get too close to the edge of the screen:

<img width="298" alt="SCR-20240718-bdvv" src="https://github.com/user-attachments/assets/31697d66-2d54-4461-9377-410821b6c3ce">

7. https://deploy-preview-13859--material-ui-x.netlify.app/x/react-charts/zoom-and-pan/#zooming-options Sentence case
8. I'm not sold on the color contrast in dark mode https://deploy-preview-13859--material-ui-x.netlify.app/x/react-charts/zoom-and-pan/#zooming-options, both series looks hard to separate.

<img width="592" alt="SCR-20240718-beyk" src="https://github.com/user-attachments/assets/b64b277c-576f-445b-a454-39649f4e63e6">

9. The zoom feature is very cool. However, I'm confused about the UX. Moved to https://github.com/mui/mui-x/issues/12503#issuecomment-2234473490.

10. Ads should be disabled on https://deploy-preview-13859--material-ui-x.netlify.app/x/react-charts/heatmap/, https://deploy-preview-13859--material-ui-x.netlify.app/x/react-charts/zoom-and-pan/

alexfauquette commented:
1. Done. The `Unstable` is useless since we are in alpha
2. Using the `import '@mui/x-charts-pro/typeOverloads';` in the `index.ts` is not enough. To fix TS, I needed to add it to the `Heatmap/index.ts`. I don't know where else to put that
3. Added "Preview" badge to [zoom and pan](https://deploy-preview-13859--material-ui-x.netlify.app/x/react-charts/zoom-and-pan/) link
4. I added charts to the [licencing page](https://deploy-preview-13859--material-ui-x.netlify.app/x/introduction/licensing/#pro-plan). @joserodolfofreitas should we rewrite some sections?
5. The JS file isn't converted -> can't do a lot about that. It's related to  https://github.com/mui/material-ui/issues/41754
6. The popover might be able to get too close to the edge of the screen.
  I don't get the point? The issue I see is the left axis not getting enough space, but no problem with the popover
  
7. https://deploy-preview-13859--material-ui-x.netlify.app/x/react-charts/zoom-and-pan/#zooming-options Sentence case

I assume it's the "O" of "Zooming Options"

8. I'm not sold on the color contrast in dark mode https://deploy-preview-13859--material-ui-x.netlify.app/x/react-charts/zoom-and-pan/#zooming-options, both series looks hard to separate.

We can change them, but in a major since it's the default colors of the charts

9. The zoom feature is very cool. However, I'm confused about the UX. Moved to https://github.com/mui/mui-x/issues/12503#issuecomment-2234473490.

10. Ads have been disabled

oliviertassinari commented:
Looks great

> 6. The popover might be able to get too close to the edge of the screen. -> I don't get the point? The issue I see is the left axis not getting enough space, but no problem with the popover

I have this in mind https://github.com/mui/base-ui/issues/39. But maybe we can't fix this yet.
